temporal-sdk/src/test/java/io/temporal/worker/ResourceBasedTunerTests.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker;
+
+import static io.temporal.testing.internal.SDKTestWorkflowRule.NAMESPACE;
+
+import com.uber.m3.tally.RootScopeBuilder;
+import com.uber.m3.util.ImmutableMap;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.common.reporter.TestStatsReporter;
+import io.temporal.serviceclient.MetricsTag;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.worker.tuning.*;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+public class ResourceBasedTunerTests {
```

Comment:
Can we include local activities in this test? I guess the resource based tuner does not emit any metrics about how many slots are currently alllocated?

Reply:
I actually need to add the `worker_task_slots_used` metric that exists in Core anyway, I have added it in my pending Go changes but didn't add it here so I'll use it from this tests and do LAs.